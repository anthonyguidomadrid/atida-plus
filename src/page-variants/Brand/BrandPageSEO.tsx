import { FunctionComponent } from 'react'
import { Brand } from 'schema-dts'
import { jsonLdScriptProps } from 'react-schemaorg'
import Head from 'next/head'
import getConfig from 'next/config'
import { useRouter } from 'next/router'

type BrandProps = {
  name?: string
  description?: string
  url?: string
  image?: string
}

export const BrandPageSEO: FunctionComponent<BrandProps> = ({
  name,
  description,
  url,
  image
}) => {
  const { publicRuntimeConfig } = getConfig()
  const host = publicRuntimeConfig.host
  const { locale } = useRouter()

  return (
    <Head>
      <script
        data-testid="structured-data-brand-page"
        type="application/ld+json"
        {...jsonLdScriptProps<Brand>({
          '@context': 'https://schema.org',
          '@type': 'Brand',
          name,
          description,
          url: `${host}/${locale}/${url}`,
          image
        })}
      />
    </Head>
  )
}
