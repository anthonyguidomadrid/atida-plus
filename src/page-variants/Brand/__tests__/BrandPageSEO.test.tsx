import { render } from '@testing-library/react'
import { BrandPageSEO } from '../BrandPageSEO'

describe(BrandPageSEO, () => {
  it('renders script tag with json-ld data', () => {
    const { container } = render(
      <BrandPageSEO
        name="Some brand"
        description="A description of the brand"
        url="brand/some-brand"
        image="some-brand-logo.png"
      />
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
