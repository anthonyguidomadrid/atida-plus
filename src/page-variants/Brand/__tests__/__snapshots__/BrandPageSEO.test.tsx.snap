// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrandPageSEO renders script tag with json-ld data 1`] = `
<script
  data-testid="structured-data-brand-page"
  type="application/ld+json"
>
  {"@context":"https://schema.org","@type":"Brand","name":"Some brand","description":"A description of the brand","url":"somehost.com/en-gb/brand/some-brand","image":"some-brand-logo.png"}
</script>
`;
